--- a/archival/libarchive/Kbuild.src
+++ b/archival/libarchive/Kbuild.src
@@ -43,7 +43,7 @@
 
 lib-$(CONFIG_AR)                        += get_header_ar.o unpack_ar_archive.o
 lib-$(CONFIG_CPIO)                      += get_header_cpio.o
-lib-$(CONFIG_TAR)                       += get_header_tar.o
+lib-$(CONFIG_TAR)                       += get_header_tar.o decompress_uncompress.o
 lib-$(CONFIG_FEATURE_TAR_TO_COMMAND)    += data_extract_to_command.o
 lib-$(CONFIG_LZOP)                      += lzo1x_1.o lzo1x_1o.o lzo1x_d.o
 lib-$(CONFIG_LZOP_COMPR_HIGH)           += lzo1x_9x.o
--- a/networking/ifplugd.c
+++ b/networking/ifplugd.c
@@ -34,16 +34,17 @@
 #include "libbb.h"
 
 #include "fix_u32.h"
-#include <linux/if.h>
-#include <linux/mii.h>
-#include <linux/ethtool.h>
 #ifdef HAVE_NET_ETHERNET_H
 # include <net/ethernet.h>
 #endif
+#include <syslog.h>
+
+#include <linux/if.h>
+#include <linux/mii.h>
+#include <linux/ethtool.h>
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
 #include <linux/sockios.h>
-#include <syslog.h>
 
 #define __user
 #include <linux/wireless.h>
