--- a/init/profile.in
+++ b/init/profile.in
@@ -36,14 +36,6 @@
     export MANPATH=$(@PKG@/libexec/addto MANPATH @PKG@/share/man)
   fi
 
-  PS_CMD=@ps@
-  if [ ! -x $PS_CMD ]; then
-      if   [ -x /bin/ps ]; then
-          PS_CMD=/bin/ps
-      elif [ -x /usr/bin/ps ]; then
-          PS_CMD=/usr/bin/ps
-      fi
-  fi
   EXPR_CMD=@expr@
   if [ ! -x $EXPR_CMD ]; then
       if   [ -x /usr/bin/expr ]; then
@@ -62,15 +54,14 @@
   fi
 
 
-  my_shell=$($PS_CMD -p $$ -ocomm=)
-  my_shell=$($EXPR_CMD    "$my_shell" : '-*\(.*\)')
+  my_shell=$($EXPR_CMD    "$0" : '-*\(.*\)')
   my_shell=$($BASENAME_CMD $my_shell)
   if [ -f @PKG@/init/$my_shell ]; then
      .    @PKG@/init/$my_shell >/dev/null # Module Support
   else
      .    @PKG@/init/sh        >/dev/null # Module Support
   fi
-  unset my_shell PS_CMD EXPR_CMD BASENAME_CMD MODULEPATH_INIT
+  unset my_shell EXPR_CMD BASENAME_CMD MODULEPATH_INIT
 fi
 
 # Local Variables:
