diff -Naur a/src/Makefile b/src/Makefile
--- a/src/Makefile	2014-10-02 15:31:36.000000000 +0000
+++ b/src/Makefile	2015-12-28 14:52:39.939842275 +0000
@@ -92,12 +92,10 @@
 scotch				:	required
 					(cd libscotch ;      $(MAKE) VERSION=$(VERSION) RELEASE=$(RELEASE) PATCHLEVEL=$(PATCHLEVEL) scotch && $(MAKE) install)
 					(cd scotch ;         $(MAKE) VERSION=$(VERSION) RELEASE=$(RELEASE) PATCHLEVEL=$(PATCHLEVEL) scotch && $(MAKE) install)
-					(cd libscotchmetis ; $(MAKE)                                                                scotch && $(MAKE) install)
 
 ptscotch			:	required
 					(cd libscotch ;      $(MAKE) VERSION=$(VERSION) RELEASE=$(RELEASE) PATCHLEVEL=$(PATCHLEVEL) ptscotch && $(MAKE) ptinstall)
 					(cd scotch ;         $(MAKE) VERSION=$(VERSION) RELEASE=$(RELEASE) PATCHLEVEL=$(PATCHLEVEL) ptscotch && $(MAKE) ptinstall)
-					(cd libscotchmetis ; $(MAKE)                                                                ptscotch && $(MAKE) ptinstall)
 
 check				:	scotch
 					(cd check ; $(MAKE) check)
@@ -112,10 +110,9 @@
 					(cd esmumps ; $(MAKE) ptscotch && $(MAKE) ptinstall)
 
 install				:	required	$(bindir)	$(includedir)	$(libdir)	$(mandir)/man1
-					-$(CP) -f ../bin/[agm]*$(EXE) $(bindir)
-					-$(CP) -f ../bin/d[agm]*$(EXE) $(bindir)
-					-$(CP) -f ../include/*scotch*.h $(includedir)
-					-$(CP) -f ../lib/*scotch*$(LIB) $(libdir)
+					-$(CP) -f ../bin/*$(EXE) $(bindir)
+					-$(CP) -f ../include/*.h $(includedir)
+					-$(CP) -f ../lib/*$(LIB) $(libdir)
 					-$(CP) -Rf ../man/* $(mandir)
 
 clean				:	required
diff -Naur a/src/libscotch/Makefile b/src/libscotch/Makefile
--- a/src/libscotch/Makefile	2014-09-23 19:28:28.000000000 +0000
+++ b/src/libscotch/Makefile	2015-12-28 14:46:12.990908897 +0000
@@ -2963,7 +2963,7 @@
 					./ptdummysizes$(EXE) library_pt_f.h ptscotchf.h
 
 libscotch$(LIB)			:	$(LIBSCOTCHDEPS)
-					$(AR) $(ARFLAGS) $(@) $(?)
+					$(AR) $(ARFLAGS) $(@) $(?) -lpthread -lz -lbz2 -lm
 					-$(RANLIB) $(@)
 
 libscotcherr$(LIB)		:	library_error$(OBJ)
