diff -Naur a/scripts/MakeHeader.py b/scripts/MakeHeader.py
--- a/scripts/MakeHeader.py	2018-02-04 18:57:13.000000000 +0000
+++ b/scripts/MakeHeader.py	2018-02-21 15:30:56.000000000 +0000
@@ -1,5 +1,6 @@
 #!/usr/bin/env python
 import os, sys, string
+import codecs
 try:
    from cStringIO import StringIO
 except ImportError:
@@ -8,6 +9,12 @@
    except ImportError:
       from io import StringIO
 
+
+def open(*args, **kwargs):
+    kwargs['encoding'] = 'utf8'
+    return codecs.open(*args, **kwargs)
+
+
 ANY=1
 COPY=2
 SKIP=3
