diff -Naur a/lddtree.sh b/lddtree.sh
--- a/lddtree.sh	2016-03-03 21:10:26.000000000 +0000
+++ b/lddtree.sh	2016-08-04 00:26:28.958252000 +0000
@@ -5,9 +5,8 @@
 
 argv0=${0##*/}
 
-: ${ROOT:=/}
+ROOT=$(realpath $mkToolchainPrefix)
 [[ ${ROOT} != */ ]] && ROOT="${ROOT}/"
-[[ ${ROOT} != /* ]] && ROOT="${PWD}${ROOT}"
 
 usage() {
 	cat <<-EOF
@@ -18,8 +17,6 @@
 	Options:
 	  -a              Show all duplicated dependencies
 	  -x              Run with debugging
-	  -R <root>       Use this ROOT filesystem tree
-	  --no-auto-root  Do not automatically prefix input ELFs with ROOT
 	  -l              Display output in a flat format
 	  -h              Show this help output
 	  -V              Show version information
@@ -45,7 +42,7 @@
 		sed -E 's: (LINUX|GNU)$: NONE:'
 }
 
-lib_paths_fallback="${ROOT}lib* ${ROOT}usr/lib* ${ROOT}usr/local/lib*"
+lib_paths_fallback="${ROOT}/lib"
 c_ldso_paths_loaded='false'
 find_elf() {
 	_find_elf=''
@@ -103,6 +100,7 @@
 							case ${line} in
 								"#"*) ;;
 								"include "*) read_ldso_conf ${line#* } ;;
+								"/"*) c_ldso_paths+=( "$line" );;
 								*) c_ldso_paths+=( "${ROOT}${line#/}" ) ;;
 							esac
 						done <"${p}"
@@ -147,7 +145,6 @@
 
 		elf_specs=$(elf_specs "${resolved}")
 		interp=$(scanelf -qF '#F%i' "${resolved}")
-		[[ -n ${interp} ]] && interp="${ROOT}${interp#/}"
 
 		if ${LIST} ; then
 			[[ -n ${interp} ]] && echo "${interp}"
@@ -158,7 +155,8 @@
 			# Extract the default lib paths out of the ldso.
 			lib_paths_ldso=$(
 				strings "${interp}" | \
-				sed -nr -e "/^\/.*lib/{s|^/?|${ROOT}|;s|/$||;s|/?:/?|\n${ROOT}|g;p}"
+				sed -nr -e "/^\/.*lib/{p}"
+				#sed -nr -e "/^\/.*lib/{s|^/?|${ROOT}|;s|/$||;s|/?:/?|\n${ROOT}|g;p}"
 			)
 		fi
 		full_interp=${interp}
@@ -209,7 +207,7 @@
 SHOW_ALL=false
 SET_X=false
 LIST=false
-AUTO_ROOT=true
+AUTO_ROOT=false
 
 while getopts haxVR:l-:  OPT ; do
 	case ${OPT} in
@@ -217,11 +215,9 @@
 	x) SET_X=true;;
 	h) usage;;
 	V) version;;
-	R) ROOT="${OPTARG%/}/";;
 	l) LIST=true;;
 	-) # Long opts ftw.
 		case ${OPTARG} in
-		no-auto-root) AUTO_ROOT=false;;
 		*) usage 1;;
 		esac
 		;;
