diff -Naur a/tools/build/src/tools/python.jam b/tools/build/src/tools/python.jam
--- a/tools/build/src/tools/python.jam	2017-12-13 23:56:50.000000000 +0000
+++ b/tools/build/src/tools/python.jam	2018-03-14 22:17:15.851096043 +0000
@@ -544,7 +544,7 @@
     }
     else
     {
-        includes ?= $(prefix)/include/python$(version) ;
+        includes ?= $(prefix)/include/python$(version)m ;
 
         local lib = $(exec-prefix)/lib ;
         libraries ?= $(lib)/python$(version)/config $(lib) ;
