diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2013-03-30 16:24:50.000000000 +0000
+++ b/CMakeLists.txt	2016-04-01 10:52:12.739525278 +0000
@@ -1,8 +1,9 @@
 cmake_minimum_required(VERSION 2.8)
 project(ParMETIS)
 
-set(GKLIB_PATH METIS/GKlib CACHE PATH "path to GKlib")
-set(METIS_PATH METIS CACHE PATH "path to METIS")
+set(GKLIB_PATH ${CMAKE_SOURCE_DIR}/metis/GKlib CACHE PATH "path to GKlib")
+set(METIS_PATH ${CMAKE_SOURCE_DIR}/metis CACHE PATH "path to METIS")
+set(SHARED FALSE CACHE BOOL "build a shared library")
 
 # Symlink ./metis to wherever metis is. This allows files to be
 # included from metis/libmetis/.
@@ -10,11 +11,7 @@
 
 # Search for MPI.
 # GK commented this out as it seems to be creating problems
-# include(FindMPI)
-# if(NOT MPI_FOUND)
-#   message(FATAL_ERROR "mpi is not found")
-# endif()
-# set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${MPI_COMPILE_FLAGS}")
+find_package(MPI REQUIRED)
 
 # Prepare libraries.
 if(SHARED)
@@ -33,7 +30,7 @@
 include_directories(${METIS_PATH}/include)
 
 # List of directories that cmake will look for CMakeLists.txt
-add_subdirectory(${METIS_PATH}/libmetis ${CMAKE_BINARY_DIR}/libmetis)
+add_subdirectory(metis)
 add_subdirectory(include)
 add_subdirectory(libparmetis)
 add_subdirectory(programs)
