diff --git a/source/base/data_out_base.cc b/source/base/data_out_base.cc
index e35adfa..6a7cb5b 100644
--- a/source/base/data_out_base.cc
+++ b/source/base/data_out_base.cc
@@ -7504,7 +7504,7 @@ DataOutBase::write_hdf5_parallel(
   AssertThrow(plist_id >= 0, ExcIO());
 #  ifdef DEAL_II_WITH_MPI
 #    ifdef H5_HAVE_PARALLEL
-  status = H5Pset_dxpl_mpio(plist_id, H5FD_MPIO_COLLECTIVE);
+  status = H5Pset_dxpl_mpio(plist_id, H5FD_MPIO_INDEPENDENT);
   AssertThrow(status >= 0, ExcIO());
 #    endif
 #  endif
