--- a/dolfin/la/PETScVector.cpp
+++ b/dolfin/la/PETScVector.cpp
@@ -749,11 +749,7 @@
 
   // Perform scatter
   VecScatter scatter;
-#if PETSC_VERSION_GE(3,11,0)
-  ierr = VecScatterCreateWithData(_x, from, _y.vec(), to, &scatter);
-#else
   ierr = VecScatterCreate(_x, from, _y.vec(), to, &scatter);
-#endif
   CHECK_ERROR("VecScatterCreate");
   ierr = VecScatterBegin(scatter, _x, _y.vec(), INSERT_VALUES,
                          SCATTER_FORWARD);
