diff -Naur a/3rdparty/CGAL-4.6/src/CMakeLists.txt b/3rdparty/CGAL-4.6/src/CMakeLists.txt
--- a/3rdparty/CGAL-4.6/src/CMakeLists.txt	2015-07-28 16:40:26.000000000 +0000
+++ b/3rdparty/CGAL-4.6/src/CMakeLists.txt	2016-04-29 08:09:56.650107730 +0000
@@ -118,7 +118,7 @@
   file(GLOB CONFIGURED_LIBS_IN_PACKAGE ${package}/src/CGAL_*/CMakeLists.txt)
   foreach (libconfigfile ${CONFIGURED_LIBS_IN_PACKAGE})
     string(REPLACE "${package}/src/" "" libconfigfile ${libconfigfile})
-    string(REPLACE "//CMakeLists.txt" "" CGAL_CONFIGURED_LIBRARY_NAME ${libconfigfile})
+    string(REPLACE "/CMakeLists.txt" "" CGAL_CONFIGURED_LIBRARY_NAME ${libconfigfile})
     if (NOT ${CGAL_CONFIGURED_LIBRARY_NAME} STREQUAL "CGAL")
 
      message(STATUS "Sources for CGAL component library '${CGAL_CONFIGURED_LIBRARY_NAME}' detected")
